controllers/api.controller.js

```diff
@@ -32,15 +32,23 @@ exports.getApiArticleById =(req,res,next)=>{
 }
 
 exports.getApiArticles=(req,res,next)=>{
-    const {sort_by="created_at",order="desc"}=req.query 
-    selectApiArticles(sort_by,order) 
+    const {sort_by="created_at",order="desc",topic}=req.query 
+
+    const checkTopic= topic?checkTopicExists(topic):Promise.resolve()
```

Comment:
This line of code really threw me off - consider installing Prettier to neaten up your code I didn't realise it was a ternary at first 😄 

I would also say the Promise.resolve is not necessary, so a simple if topic would work fine. 

Alternatively, add checkTopicExists to an array if topic is there, then use a Promise.all with selectArticles :)